### PR TITLE
Enable torch.where() to support Float16 & BFloat16 type inputs

### LINKED ISSUE #49075: Enable torch.where to support float16/bfloat16 type inputs
## 🐛 Bug

It seems that `torch.where` does not accept the input of `float16`/`bfloat16` types Tensor. 
I'm quite not sure does this is the way it should be. Maybe we can make a bit of improvement that to make `torch.where` support `float16`/`bfloat16` inputs.

## To Reproduce
```
>>> t=torch.arange(0,6).resize(2,3).to(dtype=torch.bfloat16)
>>> torch.where(t>2,torch.ones_like(t),t)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: "where_cpu" not implemented for 'BFloat16'

>>> t=torch.arange(0,6).resize(2,3).to(dtype=torch.float16)
>>> torch.where(t>2,torch.ones_like(t),t)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: "where_cpu" not implemented for 'Half'
```

## Expected behavior

```
>>> t=torch.arange(0,6).resize(2,3).to(dtype=torch.bfloat16)
>>> torch.where(t>2,torch.ones_like(t),t)
tensor([[0., 1., 2.],
        [1., 1., 1.]], dtype=torch.bfloat16)

>>> t=torch.arange(0,6).resize(2,3).to(dtype=torch.float16)
>>> torch.where(t>2,torch.ones_like(t),t)
tensor([[0., 1., 2.],
        [1., 1., 1.]], dtype=torch.float16)
```

## Environment

```
Collecting environment information...
PyTorch version: 1.8.0a0+d3b3387
Is debug build: True
CUDA used to build PyTorch: 11.0
ROCM used to build PyTorch: N/A
OS: CentOS Linux 7 (Core) (x86_64)
GCC version: (GCC) 7.5.0
Clang version: Could not collect
CMake version: version 3.14.0
Python version: 3.7 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: GeForce GTX 1050 Ti
Nvidia driver version: 450.51.05
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Versions of relevant libraries:
[pip3] numpy==1.18.5
[pip3] pytorch-sphinx-theme==0.0.24
[pip3] torch==1.8.0a0
[conda] blas                      1.0                         mkl    defaults
[conda] magma-cuda102             2.5.2                         1    pytorch
[conda] mkl                       2020.1                      217    defaults
[conda] mkl-include               2020.1                      217    defaults
[conda] mkl-service               2.3.0            py37he904b0f_0    defaults
[conda] mkl_fft                   1.1.0            py37h23d657b_0    defaults
[conda] mkl_random                1.1.1            py37h0573a6f_0    defaults
[conda] numpy                     1.18.5           py37ha1c710e_0    defaults
[conda] numpy-base                1.18.5           py37hde5b4d6_0    defaults
[conda] pytorch-sphinx-theme      0.0.24                    dev_0    <develop>
[conda] torch                     1.8.0a0                   dev_0    <develop>
```

cc @ezyang @anjali411 @dylanbespalko @mruberry @rgommers @heitorschueroff

### PR DESCRIPTION
**Pull request by Kiyosora**
Fixed #49075

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -254,8 +254,8 @@ def get_tensor(size, dtype, device, contiguous):
             height = 5
             width = 5
             for device in torch.testing.get_all_device_types():
-                for dt1 in torch.testing.get_all_dtypes(include_half=device.startswith('cuda'), include_bfloat16=False):
-                    for dt2 in torch.testing.get_all_dtypes(include_half=device.startswith('cuda'), include_bfloat16=False):
+                for dt1 in torch.testing.get_all_dtypes(include_half=True, include_bfloat16=True):
```
Comment (H-Huang) [inline test/test_torch.py:257]: We can remove these, default is True

File: test/test_torch.py [hunk 2]
```diff
@@ -6094,7 +6094,7 @@ def _where_valid_scalar_tensor_combination(self, scalar_type, dtype):
         return False
 
     @onlyOnCPUAndCUDA
-    @dtypes(*(torch.testing.get_all_int_dtypes() + torch.testing.get_all_fp_dtypes(include_bfloat16=False) +
+    @dtypes(*(torch.testing.get_all_int_dtypes() + torch.testing.get_all_fp_dtypes(include_bfloat16=True) +
```
Comment (H-Huang) [inline test/test_torch.py:6097]: Same as above

File: test/test_torch.py [hunk 3]
```diff
@@ -6106,7 +6106,7 @@ def checkRaises(scalar_type, dtype, condition, x, scalar_1):
 
         self._test_where_scalar_template(device, dtype, checkRaises)
 
-    @dtypes(*(torch.testing.get_all_int_dtypes() + torch.testing.get_all_fp_dtypes(include_bfloat16=False) +
+    @dtypes(*(torch.testing.get_all_int_dtypes() + torch.testing.get_all_fp_dtypes(include_bfloat16=True) +
```
Comment (H-Huang) [inline test/test_torch.py:6109]: Same as above

**Review by H-Huang — CHANGES_REQUESTED** [review]
Thanks for the PR! Left some feedback to remove the named arguments.

File: test/test_torch.py (hunk 1 above)
Comment (Kiyosora) [reply test/test_torch.py]: Addressed! Thanks for reminding that!

File: test/test_torch.py (hunk 2 above)
Comment (Kiyosora) [reply test/test_torch.py]: Addressed!

File: test/test_torch.py (hunk 3 above)
Comment (Kiyosora) [reply test/test_torch.py]: Addressed!

**Review by H-Huang — APPROVED** [review]
LGTM!

**Pull request closed without merging**